ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-file-answer/activityFileAnswer.component.html

```diff
@@ -1,27 +1,51 @@
+<!--  TODO: Add i18n  -->
 <ddp-question-prompt [block]="block"></ddp-question-prompt>
-<div class="file-upload-content">
+
+<div class="file-requirements" *ngIf="fileMaxSize || fileMimeTypes.length">
+    <div class="file-max-size" *ngIf="fileMaxSize">Max size: {{fileMaxSize}}</div>
+    <div class="file-types" *ngIf="fileMimeTypes.length">Allowed mime-types: {{fileMimeTypes.join(', ')}}</div>
+</div>
+
+<div class="file-upload-content" [ngClass]="{'readonly': readonly}">
     <input type="file" class="file-input"
+           [accept]="fileMimeTypes.join(',')"
            (change)="onFilesSelected($event.target.files)" #fileUpload>
 
     <div class="file-select">
-        <div class="drop-block" dropFileToUpload (filesDropped)="onFilesSelected($event)">
+        <div class="drop-block"
+             dropFileToUpload
+             (filesDropped)="!readonly && onFilesSelected($event)">
             <mat-icon class="upload-icon" color="primary">file_upload</mat-icon>
             <span class="drop-block-text">Drag files to upload</span>
         </div>
         <button mat-raised-button color="primary" class="upload-btn"
+                [disabled]="readonly"
                 (click)="fileUpload.click()">Choose file
             <mat-icon>attach_file</mat-icon>
         </button>
 
+        <div class="selected-file">{{selectedFile?.name}}</div>
+
+        <button *ngIf="!readonly" #uploadBtn
+                mat-raised-button color="primary"
+                class="submit-btn"
+                [disabled]="!isFileSelected"
+                (click)="submitFileUpload()">{{uploadedFile?.isUploaded? 'Reupload' : 'Submit'}}
+        </button>
     </div>
-    <div class="file-info">
-        <span class="file-name">{{file?.name || 'No file uploaded'}}</span>
+
+    <div class="uploaded-file">
+        <div class="title">Uploaded:</div>
+
+        <mat-chip-list>
```

Comment:
There is no any design mockup for 'File Upload' functionality.
Not sure if we are going to support multiple files uploading.
I have chosen `mat-chip` because it is already an existing material component with included remove functionality.

Reply:
I see, optionally you can consider just remove icon-button next to the file name